redis/redis.go

```diff
@@ -146,59 +146,114 @@ func (c *Client) PublishEnabled(state *EnabledMessage) {
 		})
 
 		if mErr == nil {
-			c.client.Publish(c.ctx, SyncChannelName, binMsg)
+			c.client.Do(ctx,
+				c.client.B().Publish().
+					Channel(SyncChannelName).
+					Message(rueidis.BinaryString(binMsg)).
+					Build())
 		}
 	}
 }
 
 // GetRedisCache reads the redis cache and publish it to the channel
-func (c *Client) GetRedisCache() {
+func (c *Client) GetRedisCache(ctx context.Context) {
 	c.l.Debug("GetRedisCache")
 
 	go func() {
-		iter := c.client.Scan(c.ctx, 0, prefixKey("*"), 0).Iterator()
-		for iter.Next(c.ctx) {
-			response, err := c.getResponse(iter.Val())
-			if err == nil {
-				if response != nil {
-					c.CacheChannel <- response
-				}
-			} else {
+		var cursor uint64
+
+		searchKey := prefixKey("*")
+
+		for {
+			sres, err := c.client.Do(ctx, c.client.B().Scan().Cursor(cursor).Match(searchKey).Count(1).Build()).AsScanEntry()
+			if err != nil {
 				c.l.Error("GetRedisCache ", err)
+
+				break
+			}
+
+			cursor = sres.Cursor
+
+			for _, key := range sres.Elements {
+				c.l.Debugf("GetRedisCache: %s", key)
+
+				resp := c.client.Do(ctx, c.client.B().Get().Key(key).Build())
+
+				if resp.Error() == nil {
+					str, err := resp.ToString()
+					if err == nil {
+						response, err := c.getResponse(ctx, str)
+						if err == nil {
+							c.CacheChannel <- response
+						}
+					}
+				}
```

Comment:
Are any of these errors worth logging?

Reply:
An error at this location would indicate a broken record in the database.
I guess we could log it but it's unlikely that it would procure as almost all keys are stored with a ttl.
And since the cache is usually quite big and not really maintainable for most users I deemed it as unnessecary.

Debug level would be ok I guess(?)